core/src/mindustry/world/blocks/logic/LogicDisplay.java

```diff
@@ -102,7 +103,7 @@ public void draw(){
             Draw.blend(Blending.disabled);
             Draw.draw(Draw.z(), () -> {
                 if(buffer != null){
-                    Draw.rect(Draw.wrap(buffer.getTexture()), x, y, buffer.getWidth() * Draw.scl, -buffer.getHeight() * Draw.scl);
+                    Draw.rect(Draw.wrap(buffer.getTexture()), x, y, scaleFactor * Draw.scl, -scaleFactor * Draw.scl);
```

Comment:
This should multiply rather than replace. Don't remove `buffer.getWidth()` / height.